### PR TITLE
Sentry.io integration

### PR DESCRIPTION
**Pull request by smoothcontract**
### Context
Setup integration with Sentry.io to provide production exception tracking.

### Changes proposed in this pull request
This requires new environment variable SENTRY_DSN to be set. If not present then Sentry is not activated. Also includes some additions to health endpoint - this now includes the hostname and environment name (dev / qa / uat etc.), which are also used for context in Sentry.

### Guidance to review
Ask me for sentry credentials. Relevant SENTRY_DSN has been setup on Heroku already, but also needs to be populated in Azure vault. Also need to think about how each environment will be defined so we can identify them in Sentry UI.

### CODE REVIEW

File: spec/services/health_check/environment_check_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  describe '#status' do
+    context 'with no environment name configured' do
+      it 'returns fail' do
+        Rails.configuration.environment_name = nil
+        expect(check.status).to eq :fail
+      end
+    end
+
+    context 'with host name configured' do
```
Comment (cwrw) [inline spec/services/health_check/environment_check_spec.rb]: ✂️ host/environment

Comment (smoothcontract) [reply spec/services/health_check/environment_check_spec.rb]: Good spot!

File: azure/config/ut.parameters.json [hunk 2]
```diff
@@ -40,6 +40,14 @@
         "secretName": "ut-splitApiKey"
       }
     },
+    "sentryDsn": {
+      "reference": {
+        "keyVault": {
+          "id": "/subscriptions/f35bd249-3028-48bd-9aab-d00e9d1a56b7/resourceGroups/s108d01-shared/providers/Microsoft.KeyVault/vaults/s108d01-shared-kv-01"
```
Comment (smoothcontract) [inline azure/config/ut.parameters.json]: I think this is the wrong vault actually. Oops.

File: azure/template.json [hunk 3]
```diff
@@ -118,6 +118,13 @@
         "description:": "User feedback in page Smart Survey URL"
       }
     },
+    "sentryDsn": {
+      "type": "string",
```
Comment (mafonso) [inline azure/template.json]: Is it worth it making this a `secureString'?
If this works like a token, I'd say yes.

Comment (smoothcontract) [reply azure/template.json]: Discussed offline - the value isn't something that needs to be kept exceptionally secret as there is very low potential for misuse if it was made visible on the application configuration. It's just to allow exception reporting.

**Pull request merged**